### ISSUE TITLE
Caching breaks `import/named`.

### ISSUE THREAD

**Issue by coreyfarrell** [benmosher/eslint-plugin-import#1168]
I first posted at xojs/xo/issues/335 but was told this would need to be addressed from the implementation of this rule.  Pasted from the other bug:
Create `test1.js`:
```js
export const value = 'test1';
```
Create `test2.js`:
```js
import {value} from './test1';

console.log(value);
```

Run xo, it correctly does not report any errors. Now rename the export in `test1.js`:
```js
export const value2 = 'test1';
```
Run xo again, it fails to report any errors. Run `touch test2.js && xo`, you will see the correct error:
```
  test2.js:1:9
  ✖  1:9  value not found in ./test1  import/named

  1 error
```

**Comment by benmosher** [comment]
AFAIK there is no way to fix this without changes in ESLint proper. ESLint (appropriately, per its goals) assumes that if a file is not changed, it is as correct/wrong as it was last time you scanned it.

I haven't dug into the ESLint API in ages but I assume there is still no way to do this and it would require some substantial cooperation upstream to fix permanently.

Closing as #wontfix for now because it is not possible. Would love to hear from anyone who knows I'm wrong or is willing to do the PR work with ESLint to make it possible.

**Comment by ljharb** [comment]
@benmosher right but the issue seems to be that the dependency map in this plugin is caching too aggressively?

**Comment by benmosher** [comment]
I don't think so, I think it's that 
- a imports from b
- b is changed and no longer exports what a imports
- a lints as no errors _until it is changed and ESLint re-lints it_

There is an internal cache of dependencies but it checks file modification timestamps and/or only lasts 30s or so, IIRC.
